django/db/views.py

```diff
@@ -2111,6 +2123,16 @@ def small_molecules_for_dataset(self,dataset_id):
 #        # view of the small molecule table
 #        logger.info(str(('small molecules for dataset',dataset_id,len(queryset))))
 #        return queryset
+
+    # Need a second distinction, because we'll only show the sm column 
+    # if there are datarecord entries with cells
+    def has_small_molecules_for_datarecords(self, dataset_id):
+        sql = ( 'SELECT count (smallmolecule_id) FROM db_datarecord dr '
```

Comment:
Also note that this query directly returns a boolean, so you'll also be able to drop the `> 0` from the python code.